src/main/kotlin/com/demonwav/mcdev/util/Util.kt

```diff
@@ -47,11 +46,16 @@ fun Collection<*>.toArray(): Array<Any?> {
 }
 
 @Contract(pure = true)
-inline fun <T, R> Collection<T>.mapFirstNotNull(transform: (T) -> R?): R? {
+inline fun <T, R> Iterable<T>.mapFirstNotNull(transform: (T) -> R?): R? {
     forEach { transform(it)?.let { return it } }
     return null
 }
 
+@Contract(pure = true)
+inline fun <T, R> Array<T>.mapFirstNotNull(transform: (T) -> R?): R? {
+    return asIterable().mapFirstNotNull(transform)
```

Comment:
The only thing that does is wrap the `Array` in an `Iterable`:

```kotlin
/**
 * Creates an [Iterable] instance that wraps the original array returning its elements when being iterated.
 */
public fun <T> Array<out T>.asIterable(): Iterable<T> {
    if (isEmpty()) return emptyList()
    return Iterable { this.iterator() }
}
```

Which you can see all it does is pass the `Iterator` to it. Which is _nearly identical_ to what a `for-each` loop gets compiled down to. The `only` difference between these two is the wrapping of the `Iterable`, which is subsequently only done to  pass the `Iterator`. A `for-each` loop does what it does by using the collection's `Iterator`. Compare the bytecode output for both methouds:

```kotlin
@Contract(pure = true)
inline fun <T, R> Iterable<T>.mapFirstNotNull(transform: (T) -> R?): R? {
    forEach { transform(it)?.let { return it } }
    return null
}
```
```
  // access flags 0x19
  // signature <T:Ljava/lang/Object;R:Ljava/lang/Object;>(Ljava/lang/Iterable<+TT;>;Lkotlin/jvm/functions/Function1<-TT;+TR;>;)TR;
  // declaration: R mapFirstNotNull<T, R>(java.lang.Iterable<? extends T>, kotlin.jvm.functions.Function1<? super T, ? extends R>)
  public final static mapFirstNotNull(Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;
  @Lorg/jetbrains/annotations/Contract;(pure=true) // invisible
  @Lorg/jetbrains/annotations/Nullable;() // invisible
    @Lorg/jetbrains/annotations/NotNull;() // invisible, parameter 0
    @Lorg/jetbrains/annotations/NotNull;() // invisible, parameter 1
   L0
    ALOAD 0
    LDC "$receiver"
    INVOKESTATIC kotlin/jvm/internal/Intrinsics.checkParameterIsNotNull (Ljava/lang/Object;Ljava/lang/String;)V
    ALOAD 1
    LDC "transform"
    INVOKESTATIC kotlin/jvm/internal/Intrinsics.checkParameterIsNotNull (Ljava/lang/Object;Ljava/lang/String;)V
   L1
    LINENUMBER 50 L1
    ALOAD 0
    ASTORE 3
   L2
    LINENUMBER 91 L2
    ALOAD 3
    INVOKEINTERFACE java/lang/Iterable.iterator ()Ljava/util/Iterator;
    ASTORE 4
   L3
    ALOAD 4
    INVOKEINTERFACE java/util/Iterator.hasNext ()Z
    IFEQ L4
    ALOAD 4
    INVOKEINTERFACE java/util/Iterator.next ()Ljava/lang/Object;
    ASTORE 5
   L5
    ALOAD 5
    ASTORE 6
   L6
    LINENUMBER 50 L6
    ALOAD 1
    ALOAD 6
    INVOKEINTERFACE kotlin/jvm/functions/Function1.invoke (Ljava/lang/Object;)Ljava/lang/Object;
    DUP
    IFNULL L7
    ASTORE 7
   L8
    ALOAD 7
    ASTORE 8
   L9
    LINENUMBER 50 L9
    ALOAD 8
    ARETURN
   L7
    POP
   L10
   L11
    LINENUMBER 50 L11
   L12
   L13
    GOTO L3
   L4
    LINENUMBER 92 L4
   L14
   L15
    LINENUMBER 51 L15
    ACONST_NULL
    ARETURN
   L16
    LOCALVARIABLE it Ljava/lang/Object; L9 L7 8
    LOCALVARIABLE $i$a$1$let I L9 L7 9
    LOCALVARIABLE it Ljava/lang/Object; L6 L12 6
    LOCALVARIABLE $i$a$1$forEach I L6 L12 10
    LOCALVARIABLE element$iv Ljava/lang/Object; L5 L13 5
    LOCALVARIABLE $receiver$iv Ljava/lang/Iterable; L2 L14 3
    LOCALVARIABLE $i$f$forEach I L2 L14 11
    LOCALVARIABLE $receiver Ljava/lang/Iterable; L0 L16 0
    LOCALVARIABLE transform Lkotlin/jvm/functions/Function1; L0 L16 1
    LOCALVARIABLE $i$f$mapFirstNotNull I L0 L16 2
    MAXSTACK = 2
    MAXLOCALS = 12
```

```kotlin
@Contract(pure = true)
inline fun <T, R> Array<T>.mapFirstNotNull(transform: (T) -> R?): R? {
    return asIterable().mapFirstNotNull(transform)
}
```
```
  // access flags 0x19
  // signature <T:Ljava/lang/Object;R:Ljava/lang/Object;>([TT;Lkotlin/jvm/functions/Function1<-TT;+TR;>;)TR;
  // declaration: R mapFirstNotNull<T, R>(T[], kotlin.jvm.functions.Function1<? super T, ? extends R>)
  public final static mapFirstNotNull([Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;
  @Lorg/jetbrains/annotations/Contract;(pure=true) // invisible
  @Lorg/jetbrains/annotations/Nullable;() // invisible
    @Lorg/jetbrains/annotations/NotNull;() // invisible, parameter 0
    @Lorg/jetbrains/annotations/NotNull;() // invisible, parameter 1
   L0
    ALOAD 0
    LDC "$receiver"
    INVOKESTATIC kotlin/jvm/internal/Intrinsics.checkParameterIsNotNull (Ljava/lang/Object;Ljava/lang/String;)V
    ALOAD 1
    LDC "transform"
    INVOKESTATIC kotlin/jvm/internal/Intrinsics.checkParameterIsNotNull (Ljava/lang/Object;Ljava/lang/String;)V
   L1
    LINENUMBER 56 L1
    ALOAD 0
    INVOKESTATIC kotlin/collections/ArraysKt.asIterable ([Ljava/lang/Object;)Ljava/lang/Iterable;
    ASTORE 3
   L2
    LINENUMBER 93 L2
    ALOAD 3
    ASTORE 4
   L3
    LINENUMBER 94 L3
    ALOAD 4
    INVOKEINTERFACE java/lang/Iterable.iterator ()Ljava/util/Iterator;
    ASTORE 5
   L4
    ALOAD 5
    INVOKEINTERFACE java/util/Iterator.hasNext ()Z
    IFEQ L5
    ALOAD 5
    INVOKEINTERFACE java/util/Iterator.next ()Ljava/lang/Object;
    ASTORE 6
   L6
    ALOAD 6
    ASTORE 7
   L7
    LINENUMBER 93 L7
    ALOAD 1
    ALOAD 7
    INVOKEINTERFACE kotlin/jvm/functions/Function1.invoke (Ljava/lang/Object;)Ljava/lang/Object;
    DUP
    IFNULL L8
    ASTORE 8
   L9
    ALOAD 8
    ASTORE 9
   L10
    LINENUMBER 93 L10
    ALOAD 9
    GOTO L11
   L8
    POP
   L12
   L13
    LINENUMBER 93 L13
   L14
   L15
    GOTO L4
   L5
    LINENUMBER 95 L5
   L16
    LINENUMBER 96 L16
    ACONST_NULL
   L11
    ARETURN
   L17
    LOCALVARIABLE it$iv Ljava/lang/Object; L10 L8 9
    LOCALVARIABLE $i$a$1$let I L10 L8 10
    LOCALVARIABLE it$iv Ljava/lang/Object; L7 L14 7
    LOCALVARIABLE $i$a$1$forEach I L7 L14 11
    LOCALVARIABLE element$iv$iv Ljava/lang/Object; L6 L15 6
    LOCALVARIABLE $receiver$iv$iv Ljava/lang/Iterable; L3 L16 4
    LOCALVARIABLE $i$f$forEach I L3 L16 12
    LOCALVARIABLE $receiver$iv Ljava/lang/Iterable; L2 L11 3
    LOCALVARIABLE $i$f$mapFirstNotNull I L2 L11 13
    LOCALVARIABLE $receiver [Ljava/lang/Object; L0 L17 0
    LOCALVARIABLE transform Lkotlin/jvm/functions/Function1; L0 L17 1
    LOCALVARIABLE $i$f$mapFirstNotNull I L0 L17 2
    MAXSTACK = 2
    MAXLOCALS = 14
```